getting-started/optional_installation_settings.md

```diff
@@ -0,0 +1,220 @@
+{% endhint %}
+
+Next, we will need to navigate to `config/application.rb` and customize it to look like the following:
+
+{% code title="config/application.rb" %}
+```ruby
+module AmazingStore
+  class Application < Rails::Application
+
+    # … everything else
+
+    # Don't initialize this if factory_bot_rails is not part of the current bundle group.
+    if defined?(FactoryBotRails)
+      initializer after: "factory_bot.set_factory_paths" do
+        require 'spree/testing_support'
+
+        # The paths for solidus factories.
+        solidus_paths = Spree::TestingSupport::FactoryBot.definition_file_paths
+
+        # Optional: The paths for extensions may be in different places, please refer to each extension
+        # and add the factories you require without the ".rb" extension, avoid the directory
+        # as it's supported by FactoryBot but can generate ambigous paths when a file with the
+        # same name exists.
+        extension_paths = []
+        # Optional:
+        # [ alt_1, alt_2, alt_3].map { |path| path.chomp(".rb") }
+
+        # Where:
+        # alt_1: MySolidusExtension::Engine.root.join("lib/solidus_content/factories.rb),
+        # alt_2: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product.rb"),
+        # alt_3: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product_factory.rb"),
+        # etc.
+
+        # The application factories, according to the place in which they're stored.
+        app_paths = [
+          Rails.root.join('lib', 'factories'),
+          Rails.root.join('spec', 'factories'),
+        ]
+
+        FactoryBot.definition_file_paths = solidus_paths + extension_paths + app_paths
+      end
+    end
+  end
+end
+```
+{% endcode %}
+
+Finally, modify the configuration to reference the customized methods which will give you the ability to create/build factories and stub authentication:
+
+{% code title="spec/spec\_helper.rb" %}
+```ruby
+require 'spree/testing_support/authorization_helpers'
+
+RSpec.configure do |config|
+  #...
+  config.include FactoryBot::Syntax::Methods
+end
+```
+{% endcode %}
+
+### Byebug
+
+[Byebug](https://github.com/deivid-rodriguez/byebug) is debugging tool which will help you investigate and understand code while it executes. There is no other installation instructions required other than to simply insert `byebug` method anywhere in the code base while you run a test or navigate through your development server. This will pause the execution of code where the method was inserted and provide you with a debugging interface in your shell. We highly recommend familiarizing yourself with byebug's [guide](https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md) and [commands](https://github.com/deivid-rodriguez/byebug#byebugs-commands) to learn how to get the most out of this powerful tool.
```

Comment:
```suggestion
[Byebug](https://github.com/deivid-rodriguez/byebug) is debugging tool which will help you investigate and understand code while it executes. There is no installation instructions required other than to simply insert `byebug` method anywhere in the codebase while you run a test or navigate through your development server. This will pause the execution of code where the method was inserted and provide you with a debugging interface in your shell. We highly recommend familiarizing yourself with byebug's [guide](https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md) and [commands](https://github.com/deivid-rodriguez/byebug#byebugs-commands) to learn how to get the most out of this powerful tool.
```